### ISSUE TITLE
Simplify/ease the User serialization requirement

### ISSUE THREAD

**Issue by weaverryan** [symfony/symfony#11155]
*(this issue is part of the ["DX" ("Developer eXperience") initiative](http://symfony.com/blog/making-the-symfony-experience-exceptional) introduced by Symfony project)*

In most cases, your `User` object needs to implement `Serializable` - we cover it in the docs: http://symfony.com/doc/current/cookbook/security/entity_provider.html#the-data-model (search `What is the importance of serialize and unserialize?`). If you *don't* do this, you may get odd errors if you're using Doctrine.

Could we make this more automatic? The fields that are serialized are important for [AbstractToken::hasUserChanged](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Security/Core/Authentication/Token/AbstractToken.php#L243), and unless the developer chooses to explicitly implement `EquatableInterface`, we know exactly what fields need to be serialized.

In short, the problem is well-defined and is really an ugly implementation detail. So I want to see if we can do this for them, perhaps with an optional `AbstractUser`.

**Comment by stof** [comment]
One thing we could do is write a warning in the log file in the EntityUserProvider if they use a entity which does not implement Serializable. The actual issue is that serializing a Doctrine entity can create weird issues because of proxies in case you don't implement Serializable to control the serialization. As soon as you implement Serializable, there is no issue in the provider (which is why FOSUserBundle is not affected by the issues reported by people using their own entities).

As a side note, this is not specific to the Security component. As soon as you want to be able to serialize your Doctrine entity safely, you should make it implement Serializable. Otherwise, you are likely to face an issue at some point because of a proxy getting in the middle.

**Comment by fabpot** [comment]
Closing as I do not see anything we can do to really simplify user's code.
